tests/test_invites.py

```diff
@@ -90,6 +90,27 @@ def test_invited_email_address_obfuscation(self):
 
         self.assertEqual(redacted_address, "...@1...")
 
+        # Try using a username separator string
+        self.sydent.third_party_invite_username_separator_string = "-"
+
+        email_address = "johnathon-jingle-smithington@company-town.notarealtld"
+        redacted_address = store_invite_servlet.redact_email_address(email_address)
+        # Each individual component of the username should be obfuscated, but not the domain
+        self.assertEqual(redacted_address, "johnat...-jin...-smithi...@company-...")
+
+        # Try one with a separator at a word boundary
+        self.sydent.third_party_invite_username_separator_string = "."
+
+        email_address = "applejack.@someexample.com"
+        redacted_address = store_invite_servlet.redact_email_address(email_address)
+        self.assertEqual(redacted_address, "applej...@someexam...")
+
+        # Try one where a separator is the username
+        self.sydent.third_party_invite_username_separator_string = "."
+
+        email_address = ".@someexample.com"
+        redacted_address = store_invite_servlet.redact_email_address(email_address)
+        self.assertEqual(redacted_address, ".@someexam...")
```

Comment:
It would be nice to add a test with a double separator (`--`) and what the expectation is there.